File: src/main/java/com/parrit/transformers/PersonTransformer.java
package com.parrit.transformers;

import com.parrit.DTOs.PersonDTO;
import com.parrit.entities.Person;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class PersonTransformer {

    public static PersonDTO transform(Person person) {
        return new PersonDTO(person.getId(), person.getName());
    }

    public static List<PersonDTO> transform(List<Person> persons) {
        if(persons == null || persons.isEmpty()) return Collections.emptyList();
        return persons.stream()
                .map(PersonTransformer::transform)
                .collect(Collectors.toList());
    }

    public static Person reverse(PersonDTO personDTO) {
        Person person = new Person(personDTO.getName());
        person.setId(personDTO.getId());
        return person;
    }

    public static List<Person> reverse(List<PersonDTO> personDTOs) {
        if(personDTOs == null || personDTOs.isEmpty()) return Collections.emptyList();
        return personDTOs.stream()
            .map(PersonTransformer::reverse)
            .collect(Collectors.toList());
    }

}

File: src/main/kotlin/com/parrit/transformers/PersonTransformer.kt
package com.parrit.transformers

import com.parrit.DTOs.PersonDTO
import com.parrit.entities.Person

object PersonTransformer {

    fun transform(person: Person): PersonDTO {
        val personDTO = PersonDTO()
        personDTO.id = person.id
        personDTO.name = person.name
        return personDTO
    }

    fun transform(persons: List<Person>?): List<PersonDTO> {
        when {
            persons == null || persons.isEmpty() -> return emptyList()
            else -> return persons.map { transform(it) }
        }
    }

    fun reverse(personDTO: PersonDTO) = Person(id = personDTO.id, name = personDTO.name)

    fun reverse(personDTOs: List<PersonDTO>?): MutableList<Person> {
        when {
            personDTOs == null || personDTOs.isEmpty() -> return mutableListOf()
            else -> return personDTOs.map { reverse(it) }.toMutableList()
        }
    }

}
